plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqCodeTest/controller/CodeTestChatController.kt

```diff
@@ -230,9 +236,15 @@ class CodeTestChatController(
                 }
                 .build()
 
-            val messageContent = "<span style=\"color: #EE9D28;\">&#9888;<b> ${fileInfo.fileLanguage.languageId} is not a " +
-                "language I support specialized unit test generation for at the moment.</b><br></span>The languages " +
-                "I support now are Python and Java. I can still provide examples, instructions and code suggestions."
+            val messageContent = if (fileInfo.filePath.startsWith(projectRoot.toString())) {
+                "<span style=\"color: #EE9D28;\">&#9888;<b> ${fileInfo.fileLanguage.languageId} is not a " +
+                    "language I support specialized unit test generation for at the moment.</b><br></span>The languages " +
+                    "I support now are Python and Java. I can still provide examples, instructions and code suggestions."
+            } else {
+                "<span style=\"color: #EE9D28;\">&#9888;<b>External File Detected: ${fileInfo.fileName} \n" +
+                    " is outside the project directory.</b><br></span>" +
+                    "However, I can still help you create unit tests for ${fileInfo.fileName} here"
```

Comment:
```suggestion
                "<span style=\"color: #EE9D28;\">&#9888;<b> I can't generate tests for ${fileInfo.fileName} because it's outside the project directory. I can still provide examples, instructions and code suggestions."
```